README.rst

```diff
@@ -21,23 +21,44 @@
 The async_generator library
 ===========================
 
-Python 3.6 added `async generators
-<https://www.python.org/dev/peps/pep-0525/>`__. (What's an async
-generator? `Check out my 5-minute lightning talk demo from PyCon 2016
-<https://youtu.be/PulzIT8KYLk?t=24m30s>`__.) Python 3.7 adds some more
-tools to make them usable, like ``contextlib.asynccontextmanager``.
+Many Python programmers already know and love `generators
+<https://medium.freecodecamp.org/how-and-why-you-should-use-python-generators-f6fb56650888>`__,
+because they're a super convenient way to write your own iterators
+that you can loop over using ``for``. Python 3.6 added a new thing:
+`async generators <https://www.python.org/dev/peps/pep-0525/>`__. Just
+like regular generators, they're super convenient; the difference is
+that now you loop over them using ``async for``, and this means you
+can use ``await`` inside them. If you're not doing async programming,
```

Comment:
I'm worried this might not help the confusion between *regular iterators*, *regular generators*, *async iterators* (as introduced in 3.5 / PEP 492) and *async generators* (as introduced in 3.6 / PEP 525). This is not the place to go that deep in explanations. But since you're already giving pointers, it might be helpful to clarify that—just like generators are a nice way to write iterators—async generators are super useful to define async iterators.